llvm/include/llvm/IR/VectorTypeUtils.h

```diff
@@ -71,6 +75,18 @@ inline bool isVectorizedTy(Type *Ty) {
   return Ty->isVectorTy();
 }
 
+// Returns true if `Ty` is a valid vector element type, void, or an unpacked
+// literal struct where all elements are valid vector element types.
+// Note: Even if a type can be vectorized that does not mean it is valid to do
+// so in all cases. For example, a vectorized struct (as returned by
+// toVectorizedTy) does not perform (de)interleaving, so it can't be used for
+// vectorizing loads/stores.
+inline bool canVectorizeTy(Type *Ty) {
```

Comment:
IIUC the meaning of `canVectorize` here is specific to LoopVectorize and only used there, so it would make sense to keep it there?

Reply:
Ah right, thanks, I guess I got confused mixing the terminology.